docs: organize docs directory (#4367)
carbon-design-system/carbon: docs/README.md

# Carbon documentation

> Welcome to the Carbon documentation. Here you'll find information on project

Reviewer: ```suggestion
This is the home for all process, contribution, and version-related documentation. These docs include information on project
```

Author: @connor-leech Why call out "not related to tooling"?

Reviewer: Because this isn't an exhaustive repository of our documentation, it's primarily docs that don't live on our website.

Author: Do we have "tooling" documentation elsewhere?

Reviewer: Our website? How to install Sketch, how to use our grid, icons, etc. Component usage guidance. 

If you think there's a better word than tooling I'd prefer it, as I typically don't like using that word.

Author: The /docs folder is for process, contribution, and version-related documentation. The website is usage guidelines and resources list.

Reviewer: Brilliant, just updated 👍

> management, contribution, migration, lessons learned along the way, etc.

Reviewer: ```suggestion
management, contribution, migration, reflections, and process insights.

If you think something is missing, we encourage [contributions](#how-to-contribute).
```

## Project management

- [Endgame](/project-management/endgame.md)
- [Versioning](/project-management/versioning.md)
- [Support](/project-management/support.md)

> ## Contributing

Reviewer: ```suggestion
## How to contribute
```
These headers should reflect those on our site

# Carbon documentation

This is the home for all process, contribution, and version-related documentation. These docs include information on project
management, contribution, migration, reflections, and process insights.

If you think something is missing, we encourage [contributions](#how-to-contribute).

## Project management

- [Endgame](/project-management/endgame.md)
- [Versioning](/project-management/versioning.md)
- [Support](/project-management/support.md)

## How to contribute

> - [Developer handbook](/contributing/developer-handbook.md)

Reviewer 2: Could you talk through some of the groupings? Not sure I understand all of them.

For example, Windows HCM is more related to support and definition and done over contributing. IBM Plex is more of a guide for using IBM Plex versus something folks do when working with the project
